src/github-cli/install.sh

```diff
@@ -124,138 +124,85 @@ check_packages() {
-        else
-            ((breakfix=breakfix-1))
-            if [ "${breakfix}" = "0" ] && [ "${last_part_optional}" = "true" ]; then
-                declare -g ${variable_name}="${major}.${minor}"
-            else 
-                declare -g ${variable_name}="${major}.${minor}.${breakfix}"
-            fi
+    local prefix="${1:-}"
+    git ls-remote --tags --sort=-version:refname https://github.com/cli/cli.git | while read -r _ ref; do
+        version="${ref#refs/tags/v}"
+        if [[ $version != *-* && ( -z $prefix || "${version}." == "${prefix}."* ) ]]; then
+            printf "%s\n" "$version"
         fi
-    set -e
+    done
 }
 
-# Fall back on direct download if no apt package exists
-# Fetches .deb file to be installed with dpkg
+# For each version, fetch a .deb file from releases and install the first available one with dpkg
 install_deb_using_github() {
-    check_packages wget
-    arch=$(dpkg --print-architecture)
-
-    find_version_from_git_tags CLI_VERSION https://github.com/cli/cli
-    cli_filename="gh_${CLI_VERSION}_linux_${arch}.deb"
-
-    mkdir -p /tmp/ghcli
-    pushd /tmp/ghcli
-    wget https://github.com/cli/cli/releases/download/v${CLI_VERSION}/${cli_filename}
-    exit_code=$?
-    set -e
-    if [ "$exit_code" != "0" ]; then
-        # Handle situation where git tags are ahead of what was is available to actually download
-        echo "(!) github-cli version ${CLI_VERSION} failed to download. Attempting to fall back one version to retry..."
-        find_prev_version_from_git_tags CLI_VERSION https://github.com/cli/cli
-        wget https://github.com/cli/cli/releases/download/v${CLI_VERSION}/${cli_filename}
-    fi
-
-    dpkg -i /tmp/ghcli/${cli_filename}
-    popd
+    local arch="$(dpkg --print-architecture)"
+    local version download_url
+    for version; do
+        download_url="https://github.com/cli/cli/releases/download/v${version}/gh_${version}_linux_${arch}.deb"
+        wget -P /tmp/ghcli "$download_url" || continue
+        dpkg -i "/tmp/ghcli/$(basename "$download_url")"
+        break
+    done
     rm -rf /tmp/ghcli
 }
 
-export DEBIAN_FRONTEND=noninteractive
-
-# Install curl, apt-transport-https, curl, gpg, or dirmngr, git if missing
-check_packages curl ca-certificates apt-transport-https dirmngr gnupg2
-if ! type git > /dev/null 2>&1; then
-    check_packages git
+if [ -n "$TEST_EXPORT" ]; then
+    # print function named by TEST_EXPORT and exit
+    declare -f "$TEST_EXPORT"
+    exit 0
 fi
 
-# Soft version matching
-if [ "${CLI_VERSION}" != "latest" ] && [ "${CLI_VERSION}" != "lts" ] && [ "${CLI_VERSION}" != "stable" ]; then
-    find_version_from_git_tags CLI_VERSION "https://github.com/cli/cli"
-    version_suffix="=${CLI_VERSION}"
-else
-    version_suffix=""
+check_packages ca-certificates
+type -P git >/dev/null || check_packages git
+
+is_latest="false"
+
+case "$CLI_VERSION" in
+none )
+    # opt out of installing gh
+    exit 0
+    ;;
+latest | lts | stable | current )
```

Comment:
Is my interpretation of `"version": "none"` and `"version": "current"` correct? I wasn't sure how those values are supposed to be handled since they aren't documented

Reply:
The current `proposals` property mentions `latest` and `none`. But if the gihutb-cli supports other keywords, then it's nice to add. Thanks! 

Feel free to update the proposals here as well - [here](https://github.com/devcontainers/features/blob/e7ebc0e9de1e3cf61424096352561d94b7e3e9a0/src/github-cli/devcontainer-feature.json#L10)

Reply:
From existing code, it looked like `github-cli` feature supported many different keywords, but I only want to support the old ones for backwards compatibility and not actively suggest their use. I do not even understand what `none` is about. The usages I'd like to document are:

1. `"version": "latest"` - default
2. `"version": "2.20.1"` - lock to a specific version
3. `"version": "2.20"` - install the latest 2.20.x
4. `"version": "2"` - install the latest 2.x

Any suggestions how to best document this?

Reply:
I see, for `version` option we are accepting semver values so I think it should be self explanatory of how versions are computed. However, it's great to document and make the usage clearer. Feel free to add a `NOTES.md` file in https://github.com/devcontainers/features/tree/main/src/github-cli. The action would automatically add the contents to the [README](https://github.com/devcontainers/features/blob/main/src/github-cli/README.md) file.

(example - [docker-in-docker](https://github.com/devcontainers/features/tree/main/src/docker-in-docker))